lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -278,11 +284,16 @@ export default function getRecipeInterface(
                     response.session.userId,
                     response.session.userDataInJWT,
                     res,
-                    req
+                    req,
+                    transferMethod
                 );
 
                 return session;
             } catch (err) {
+                if (err.type === STError.TRY_REFRESH_TOKEN && sessionOptional && !gotRID && method !== "get") {
+                    return undefined;
+                }
```

Comment:
I don't quite understand this part. Session optional is a boolean that has effect when we get UNAUTHORISED error and not try refresh token.

Reply:
Also i don't understand why && !gotRID && method !== "get" matters.